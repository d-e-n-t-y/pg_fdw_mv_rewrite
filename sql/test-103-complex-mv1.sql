explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test_lb where hidden = 'hidden2' group by key;

select key, COUNT (value) from test_lb where hidden = 'hidden2' group by key;
