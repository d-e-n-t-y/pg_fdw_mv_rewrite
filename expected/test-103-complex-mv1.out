-- Anticipate no rewrite
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test where hidden = 'hidden2' group by key;
                            QUERY PLAN                             
-------------------------------------------------------------------
 Finalize GroupAggregate
   Output: key, count(value)
   Group Key: test.key
   ->  Gather Merge
         Output: key, (PARTIAL count(value))
         Workers Planned: 2
         ->  Partial GroupAggregate
               Output: key, PARTIAL count(value)
               Group Key: test.key
               ->  Sort
                     Output: key, value
                     Sort Key: test.key
                     ->  Parallel Seq Scan on public.test
                           Output: key, value
                           Filter: (test.hidden = 'hidden2'::text)
(15 rows)

select key, COUNT (value) from test where hidden = 'hidden2' group by key;
 key  |  count  
------+---------
 key2 | 1000000
(1 row)

-- Anticipate rewrite
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test where hidden = 'hidden2' group by key, hidden;
              QUERY PLAN               
---------------------------------------
 Custom Scan (MVRewriteScan)
   Output: key, (count(value)), hidden
   Rewritten: scan of public.test_mv2
   ->  Seq Scan on public.test_mv2
         Output: key, count, hidden
(5 rows)

select key, COUNT (value) from test where hidden = 'hidden2' group by key, hidden;
 key  |  count  
------+---------
 key2 | 1000000
(1 row)

